tests/Agent/IntegrationTests/IntegrationTests/RemoteServiceFixtures/AwsLambda/LambdaTestToolFixture.cs

```diff
@@ -64,13 +65,33 @@ private void AddAdditionalEnvironmentVariableIfNotNull(string name, string value
             }
         }
 
-        private string WarmUpTestTool()
+        private void WarmUpTestTool()
         {
-            var address = $"http://localhost:{LambdaTestTool.Port}";
 
-            TestLogger?.WriteLine($"[LambdaTestToolFixture] Warming up collector via: {address}");
+            for (var attempt = 1; attempt <= 3; attempt++)
+            {
+                try
+                {
+                    var address = $"http://localhost:{LambdaTestTool.Port}";
+
+                    TestLogger?.WriteLine($"[LambdaTestToolFixture] Warming up lambda test tool via: {address} attempt {attempt}.");
+
+                    GetString(address);
+                    return;
+                }
+                catch(Exception e)
+                {
+                    TestLogger?.WriteLine($"Unable to warm up lambda test tool during attempt {attempt}. Exception: {e}");
+                    if (attempt == 3)
+                    {
+                        // Halt the test because the test tool was unable to start
+                        throw;
+                    }
 
-            return GetString(address);
+                    // Wait a little to give the test tool more time to start
+                    Thread.Sleep(TimeSpan.FromSeconds(10));
```

Comment:
I would hope 30 seconds would be enough, but do you know how long it typically takes on GitHub?

Reply:
Unfortunately I don't have an estimate for how long it typically takes. I was hoping to see if this change made a difference and then adjust it necessary. When the tests succeed, we do not get any of the test output so I can't see how many retries occurred.